addons/mail/static/tests/qunit_suite_tests/models/file_uploader_tests.js

```diff
@@ -50,6 +53,70 @@ QUnit.test("no conflicts between file uploaders", async function () {
     await contains(".o_Chatter .o_AttachmentCard");
     await contains(".o_ChatWindow .o_Message .o_AttachmentCard");
 });
+
+QUnit.test('Arrows on nonviewable attachment', async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId = pyEnv['res.partner'].create({});
+    const irAttachmentId = pyEnv['ir.attachment'].create({
+        mimetype: 'text/plain',
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+    pyEnv['mail.message'].create({
+        attachment_ids: [irAttachmentId],
+        model: 'res.partner',
+        res_id: resPartnerId,
+    });
+    pyEnv['res.partner'].write([resPartnerId], {message_main_attachment_id : irAttachmentId})
+    const views = {
+        'res.partner,false,form':
+            '<form string="Partners">' +
+                '<sheet>' +
+                    '<field name="name"/>' +
+                '</sheet>' +
+                '<div class="o_attachment_preview"/>' +
+                '<div class="oe_chatter">' +
+                    '<field name="message_ids"/>' +
+                '</div>' +
+            '</form>',
+    };
+    patchUiSize({ size: SIZES.XXL });
+    const { messaging, openFormView } = await start({
+        mockRPC(route, args) {
+            if (_.str.contains(route, '/web/static/lib/pdfjs/web/viewer.html')) {
+                var canvas = document.createElement('canvas');
+                return canvas.toDataURL();
+            }
+        },
+        serverData: { views },
+    });
+    await openFormView({
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+
+    // No viewer as text file is non viewable
+    await contains(".oe_chatter");
+    await contains(".o_attachment_preview", { count: 0 });
+    await contains(".o_attachment_preview_container > iframe", { count: 0 });
+    await contains(".arrow", { count: 0 });
+
+    // Add a PDF file
+    await click(".o_ChatterTopbar_buttonSendMessage");
+    await contains(".o_ComposerTextInput_textarea");
+    inputFiles(messaging.models["Chatter"].all()[0].composerView.fileUploader.fileInput, [
+        await createInputFile({ name: "invoice.pdf", contentType: "application/pdf" }),
+    ]);
+    await click(".o_Composer_buttonSend:enabled");
+    await contains(".o_attachment_preview_container > iframe", { count: 0 }); // The viewer tries to display the text file not the PDF
+    await contains(".arrow", { count: 2 }); // Display arrows to switch to the viewable attachment
+
+    // Switch to the PDF file in the viewer
+    await click(".o_move_next");
+    await contains(".o_attachment_preview_container > iframe"); // There should be iframe for PDF viewer
+    await contains(".arrow", { count: 0 }); // Hide arrows as the text file is not viewable
+});
+
```

Comment:
No need to assert `.arrow { count: 2 }`: you click on `.o_move_next` anyway which already implicitly assert presence of this interaction.

I wouldn't both asserting no arrow at the end. What matters in fix is that you can select a viewable attachment, let's not test too much because we are not necessarily proud of the limitation and strange things like stop displaying arrow. This also limits the amount of testing to what really matters and let us improve it without having to maintain lots of test that were too greedy.